torchtitan/distributed/utils.py

```diff
@@ -188,17 +190,13 @@ def context(cp_context: Generator[None, None, None] | None = None):
                 )
 
             if cp_context is not None:
-                from torch.nn.attention import sdpa_kernel, SDPBackend
-
-                stack.enter_context(
-                    sdpa_kernel(
-                        [
-                            SDPBackend.FLASH_ATTENTION,
-                            SDPBackend.EFFICIENT_ATTENTION,
-                            SDPBackend.CUDNN_ATTENTION,
-                        ]
-                    )
-                )
+                try:
+                    ScaledDotProductAttention.backends.remove(SDPBackend.MATH)
+                except ValueError:
+                    pass
```

Comment:
nit: try-catch sounds a bit heavy, let's just
```suggestion
                if SDPBackend.MATH in ScaledDotProductAttention.backends:
                    ScaledDotProductAttention.backends.remove(SDPBackend.MATH)
```

Reply:
EAFP is actually a preferred way for many Python documents. But I know people have different opinions. Let me change it.

Reply:
Oh, I didn't even know about LBYL vs. EAFP. I learnt from https://realpython.com/python-lbyl-vs-eafp/
I think either is OK then.

Reply:
I don't have a strong preference. I think the most important thing is consistency through out the repo. I checked the repo, I believe we are using LBYL more. So let me change it.